src/sql/workbench/parts/notebook/models/notebookModel.ts

```diff
@@ -292,7 +297,20 @@ export class NotebookModel extends Disposable implements INotebookModel {
 				this._defaultLanguageInfo = this.getDefaultLanguageInfo(contents);
 				this._savedKernelInfo = this.getSavedKernelInfo(contents);
 				if (contents.cells && contents.cells.length > 0) {
-					this._cells = contents.cells.map(c => factory.createCell(c, { notebook: this, isTrusted: isTrusted }));
+					this._cells = contents.cells.map(c => {
+						let cellContent = (<nb.ICellContents>c);
+						if (c && cellContent && cellContent.cell_type === CellTypes.Markdown) {
+							this._textCellsLoading++;
+						}
+						let cellModel = factory.createCell(c, { notebook: this, isTrusted: isTrusted });
+						this._register(cellModel.onLoaded((e) => {
+							this._textCellsLoading--;
+							if (this._textCellsLoading <= 0) {
+								this.telemetryService.publicLog(TelemetryKeys.NotebookMarkdownRendered);
+							}
```

Comment:
You mentioned using timesincesessionstart for timing. I assume the goal is to time how much time passes from notebook load to the contents being rendered? In general I'd recommend not using timesincesessionstart because: 

1. What are you basing it against? Session start isn't great since you'd have no idea when the notebook itself is loaded. And even if there was a NotebookLoaded event there's no way to correlate them since we don't have user/machine IDs to aggregate events together. 

2. You shouldn't use timesincesessionstart for timing regardless since that's added at some point later - which may be well after the notebook has actually been rendered

Instead I'd suggest having a timer that tracks the time taken to load and pass that in as a property to publicLog

Reply:
There's a telemetry event called "telemetry/editorOpened" that is also emitted immediately when the notebook editor is opened. Subtract timesincesessionstart from these 2 events and that's the time to render the markdown cells in a notebook.

Regardless of that, if there's value in just emitting the actual number with this event, I'm totally open to that 😄

Reply:
Yes, but how do you plan on correlating those two events together? Without a user or session ID there's no way we can accurately do that. So in order to get that number you'll need to count the time yourself.

Reply:
Agreed here, would be nice to have some data that goes along with the event, so that it's more meaningful.

Reply:
@Charles-Gagnon makes sense, thanks. I was thinking more of just the perf test perspective, but totally understand here, so appreciate the feedback. I'll create a timer and output the data with this event. Thanks for confirming, @abist